Added option for minimal build

skogtake opened the pull request:
Added webpack config for building the project without certain modules: Default UI, Dash, Cast and shaka-player.

yairans committed:
playlist - the thumbnail slices request is missing ks when the item is already played (#210)

pass the current session object to `setMedia` api

yairans committed:
fix(FEC-8842): playlist - item sources config ignored (#211)

* config the player with the item sources config before `loadMedia`
* delete empty url  properties (`hls\dash\progressive`) from item sources config, otherwise the empty url property will override the provider's url property

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.38.0

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.38.1

odedhutzler committed:
chore(FEC-8572): fix failing test (#213)

* fix tests according to the new font style object
* revert karma conf changes
* removing implicitFontScale which is not part of the object

yairans committed:
feat(FEC-8703): playlist loop (#208)

add `loop` option and consider it on `PLAYBACK_ENDED` event

OrenMe committed:
fix(FEC-8834): remove scrubber preview if entry is youtube (#206)


Do not add scrubber if entry is youtube, as we can't generate it via our API

yairans committed:
fix(FEC-8833):playlist - no thumbnail preview when the item already played (#209)

use `addKalturaPoster` for array only (OTT)

Also fixing **FEC-8840**

medin-n commented on src/index.html:
```diff
@@ -24,6 +24,8 @@
     var kalturaPlayer = KalturaPlayer.setup(config);
     kalturaPlayer.loadMedia({entryId: '0_wifqaipd'});
+    document.querySelector('.kaltura-player-container').appendChild(kalturaPlayer.getView());
+    kalturaPlayer = kalturaPlayer;
```
no need for this

elsassph commented on package.json:
```diff
@@ -21,7 +23,6 @@
     "playkit-dev-ott:start": "playkit-dev start ott",
     "playkit-dev:stop": "playkit-dev stop",
     "playkit-rel": "playkit-rel start",
-    "precommit": "lint-staged",
```
Why removing linting?

medin-n replied:
![image](https://user-images.githubusercontent.com/5623935/51604469-2f4b2300-1f0d-11e9-9b0f-bae8cc23fb90.png)

elsassph replied:
Mayyybe just fix the errors? :D

medin-n replied:
reverted

Commit:
fix(FEC-8688): When pausing video in cast and then disconnect the player display first play poster and not video stream with paused point (#200)


Let the player always autoplay and pause it if necessary

Commit:
Minimal build with excluded moddules

elsassph commented on README.md:
```diff
@@ -1,36 +1,8 @@
-# Kaltura Player JS Platform - Cloud TV and OVP Media Players Based on the [PlayKit JS Player]
+# Massive Fork of Kaltura Player
```
Please revert the readme and add an extra readme for the minimal version

medin-n replied:
done

elsassph commented:
Approved.

yairans committed:
docs: add references and fix links (#216)

Commit:
Updated documentation for maintaining the fork

yairans committed:
fix(FEC-8298): SOURCE_SELECTED event triggered before the UI ready when the source provided manually (#129)

postpone the `sources` config after the ui initialization

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.39.0

yairans committed:
chore: update CHANGELOG.md

Commit:
docs: ads (#215)


Ads framework docs

yairans committed:
fix(FEC-8871): playlist - strange behavior when playback.loop is true (#217)

force `playback.loop: false` for playlist

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.39.1

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.39.2

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.39.3

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.39.4

AnnaAleksandrowicz committed:
docs: update playlist options (#219)

OrenMe committed:
docs: update invalid link

closes #198

odedhutzler committed:
feat(FEC-8769): do not allow to create two players with the same target id (#218)

* add same target id setup validation

* fix tests

Commit:
Minimal OTT player

Commit:
Merge branch 'minimal-ott' into minimal

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.40.0

Commit:
fix(FEC-8901): duplication of ClosedCaptions on iOS on none Safari browser (#221)

Changed the `if` condition of `checkNativeTextTracksSupport` function, to allow other browsers in iOS to set it to `true` by default.

From `isSafari()` -> to `isSafari() || isIos()`

- Tested the matter with iPad in Chrome, Safari, and FireFox.

odedhutzler committed:
fix(FEC-8927): player crash on Echo show browser (#223)

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.40.1

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.40.2

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.3

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.4

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.5

odedhutzler committed:
chore: update playkit-js-* versions

odedhutzler committed:
chore: update dist

odedhutzler committed:
chore(release): 0.40.6

yairans committed:
fix(FEC-8973): safari desktop plays using hlsjs instead of natively (#225)

fix Env path

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.7

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.8

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.9

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.10

yairans committed:
fix(FEC-8981): playlist doesn't send ott analytics (#226)

Pass plugins with empty config to evaluation as we do for single media (in `setMedia`)

OrenMe committed:
docs: add ads link in guides

OrenMe committed:
feat(FEC-8662): add UI config token evaluation for share (#228)


add UI config token evaluation - used for share mainly now

Yuvalke committed:
fix(FEC-8826): fullscreen implementation moved to core (#227)

Element as parameter to open in fullscreen

yairans committed:
chore: update playkit-js-* versions

yairans committed:
Merge branch 'master' of https://github.com/kaltura/kaltura-player-js

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.11

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.40.12

OrenMe committed:
fix(FEC-9002): analytics aren't being sent (#230)


The entryId is not being evaluated as the dictionary sent to the evaluation was changed and it didn't include the new entry ID.
All analytics plugins are validating that they have a valid entry ID and because they don't then no event is fired.

yairans committed:
fix(FEC-9013): [iOS] - Player doesnt play ads native (#234)

set default value in ima plugin (https://github.com/kaltura/playkit-js-ima/pull/103) instead in kaltura player.
delayInitUntilSourceSelected will be set to true always by default.

Yuvalke committed:
fix(FEC-8826): default element is video container instead main container (#235)

Add default in kaltura player to the main container

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.40.13

Yuvalke committed:
fix(FEC-9026): [Player_V3][Flash] - When clicking fullscreen icon in flash, the player area stays in the same size

Fullscreen will apply to kaltura container instead user container.

RoyBregman committed:
chore: update playkit-js-* versions

RoyBregman committed:
chore: update dist

RoyBregman committed:
chore(release): 0.40.14

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.40.15

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.41.0

yairans committed:
chore: update dist

amirch1 committed:
fix(FEC-9052): allow player to load from partner 0(#237)

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.41.1

Commit:
Merge remote-tracking branch 'kaltura-fork/master' into minimal

Commit:
playable item

Yuvalke committed:
fix(FEC-9067): playback error with ima on LG TV WebOS (#239)

add default configuration for LG TV

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.41.2

yairans committed:
fix(FEC-9087): IMA DAI - the pre-roll Ad duration also calculated for start position - the playback started from 10th sec instead of 20th (#240)

config hls by default `autoStartLoad:false` to enable https://github.com/kaltura/playkit-js-ima-dai/pull/15 to override the [`startPosition`](https://github.com/video-dev/hls.js/blob/master/docs/API.md#startposition) value.

Yuvalke committed:
fix(FEC-9177): Smart TV showed as mobile device (#242)

Change every check of device.type to mobile or tablet instead, there are more device types which not relevant.

Yuvalke committed:
fix(FEC-9118): External caption doesn't display (#243)

UA of webos v.2 is fake to safari, added check for mac os to prevent this fake.
LG check is added to UAParser.

Yuvalke committed:
fix(FEC-9103): Remove old kaltura stats (#244)

Remove kanalytics from kaltura-player.

yairans committed:
fix(FEC-8940): cannot configure plugin by array (#222)

iterating on array

yairans committed:
feat(FEC-8631): bumper plugin (#224)

Set `inBrowserFullscreenForIOS :true` by default

RoyBregman committed:
feat(FEC-8975): QoS Data enhancement (#241)

OrenMe committed:
Revert "feat(FEC-8975): QoS Data enhancement (#241)"

This reverts commit 8f3b49250a4a0f75151e70f3564295c28e740bee.

yairans committed:
fix: bumper and ima-dai integration (#245)

set bumper config 
```
disableMediaPreload: true,
position: [0]
```
to avoid conflicts with dai plugin

Yuvalke committed:
fix(FEC-9114): live doesn't work properly with hls.js (#246)

change default configuration for LG TV to play with native video player by default instead hls/shaka. Currently we use this only when playing ads.

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
Revert "chore: update playkit-js-* versions"

This reverts commit 85a955559f6962786f0e43c7d797be043327254e.

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
Revert "chore: update playkit-js-* versions"

This reverts commit c6c6f96bcdb2355b306d6e894be32390747d61ce.

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.42.0

yairans committed:
fix: inBrowserFullscreen path (#248)

set `playback.inBrowserFullscreen` instead of `ui.components.fullscreen.inBrowserFullscreenForIOS`

Yuvalke committed:
feat(FEC-9023): more logger options (#247)

add configuration logger options.
handler to output logs with custimize function.
option to remove playerVersion from log.

OrenMe committed:
feat(FEC-9024): send beacon for non partner usage (#249)


send usage beacon for usages of player without partner - this will be used to understand when player is configured incorrectly.

RoyBregman committed:
chore: update playkit-js-* versions

RoyBregman committed:
chore: update dist

RoyBregman committed:
chore(release): 0.43.0

Commit:
Merge branch 'master' into minimal

Commit:
Build minimal v0.43.0

yairans committed:
feat(FEC-9145): support non sibling video tags (#250)

set `bumper` default config according to the env and config

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore(release): 0.44.0

yairans committed:
chore: update dist

yairans committed:
feat(FEC-9156): getting bumper from playback context (#252)

Consider plugins come from the provider

Yuvalke committed:
feat(FEC-9227): support reInit of MSE (#254)

Add configuration to handle ads with MSE

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.45.0

RoyBregman committed:
chore: update playkit-js-* versions

RoyBregman committed:
chore: update dist

RoyBregman committed:
chore(release): 0.45.1

Yuvalke committed:
fix: useNativeTextTrack doesn't change to true by default on MacOS (#258)

fix check for macOS

Yuvalke committed:
fix(FEC-9259):Smart TV detect as safari browser (#259)

Add check for safari with mac os to fix Samsung detection as safari, safari plays native instead SmartTV should play with Hls and Shaka.

Yuvalke committed:
fix(FEC-9260): Autoplay is not working on LG TV (#260)

LG SDK 4 break their support on base64 source which included in our check for autoplay

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.45.2

RoyBregman committed:
chore: update playkit-js-* versions

RoyBregman committed:
chore: update dist

RoyBregman committed:
chore(release): 0.45.3

yairans committed:
fix(FEC-9273): playing preroll on AV player (ios+playsinline=false) gets the player stuck (#262)

config `disableMediaPreload:true` for ios + playsinline is false

yairans committed:
chore: update playkit-js-* versions

yairans committed:
chore: update dist

yairans committed:
chore(release): 0.45.4

Yuvalke committed:
chore: update playkit-js-* versions

Yuvalke committed:
chore: update dist

Yuvalke committed:
chore(release): 0.45.5

skogtake committed:
Merge branch 'master' of https://github.com/kaltura/kaltura-player-js into updates-from-master

skogtake committed:
added /dist build files

medin-n committed:
Merge pull request #3 from massiveinteractive/updates-from-master

Updates from master
